### PR TITLE
gui: Add Tor icon

### LINKED ISSUE #58: Tor status indicator icon in bottom right
Moved from: https://github.com/bitcoin/bitcoin/issues/7734.

The current QT GUI shows some network status indicators in the bottom right.

I'd like someone to add one for Tor. It might be the Tor onion logo that is either green, yellow (if going through some not-quite-ready stage) or with a circle and line striking through it when not connected via Tor.

This would complement the Tor-related improvements in 0.12.0 and make it easier for users to determine whether they are connected via Tor without doing additional investigatory steps, such as using the  `-debug=tor` switch.

### LINKED ISSUE #7734: Feat Request (Qt/GUI): Tor status indicator icon in bottom right
The current QT GUI shows some network status indicators in the bottom right.

I'd like someone to add one for Tor. It might be the Tor onion logo that is either green, yellow (if going through some not-quite-ready stage) or with a circle and line striking through it when not connected via Tor.

This would complement the Tor-related improvements in 0.12.0 and make it easier for users to determine whether they are connected via Tor without doing additional investigatory steps, such as using the  `-debug=tor` switch.

### PR DESCRIPTION
**Pull request by hebasto**
This PR is submitted not to the GUI repo because it touches files out of the `src/qt` directory.

Behavior:
- the Tor icon is showed in the status bar iif _all_ peer connections are made via Tor ([credits](https://github.com/bitcoin-core/gui/issues/58#issuecomment-685922056) to **Sjors**, these conditions could be achieved with `-onlynet=onion`, `-listen=0`, and `-listenonion=1` options)
- a click on the Tor icon opens the "Network" tab of the "Options" dialog

Designer: **Bosch-0**
Icon PNG file is optimized ([credits](https://github.com/bitcoin/bitcoin/commit/42f6a0c2b946af2cce86b721b349d35c4e21ce88) to **jonasschnelli**).

Fix #7734
Fix https://github.com/bitcoin-core/gui/issues/58

Screenshots:
Screenshot from 2020-09-09 17-04-56
Screenshot from 2020-09-09 17-03-54

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Would it make sense to expose the same boolean on RPC?

Btw, the latest optimize png script is now here: https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/master/optimize-pngs.py

Also, it would be good to add the svg source in case alternative versions need to be created.

**Comment by hebasto** [comment]
> Would it make sense to expose the same boolean on RPC?

With which command name?

**Comment by kristapsk** [comment]
> Would it make sense to expose the same boolean on RPC?

I think that's a good idea.

**Comment by kristapsk** [comment]
> > Would it make sense to expose the same boolean on RPC?

> With which command name?

`getnetworkinfo`?

**Comment by fanquake** [comment]
> This PR is submitted not to the GUI repo because it touches files out of the src/qt directory.

Given this is just node.cpp/h and the changes are trivial, I think this should be in the GUI repo.

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -1317,6 +1323,15 @@ void BitcoinGUI::updateProxyIcon()
...
 }
 
+void BitcoinGUI::updateTorIcon()
+{
+    if (clientModel->hasTorOnlyConnections()) {
+        m_tor_icon->setPixmap(platformStyle->SingleColorIcon(":/icons/tor_connected").pixmap(STATUSBAR_ICONSIZE, STATUSBAR_ICONSIZE));
+        m_tor_icon->setToolTip(tr("All connections are <b>via Tor</b> only"));
+        m_tor_icon->show();
+    }
+}
```
Comment (benthecarman) [inline src/qt/bitcoingui.cpp:1333]: Should this have an else case:
```
else {
	m_tor_icon->hide();
}
```

Comment (hebasto) [reply src/qt/bitcoingui.cpp:1333]: Done in https://github.com/bitcoin-core/gui/pull/86.

**Comment by hebasto** [comment]
@fanquake 
> Given this is just node.cpp/h and the changes are trivial, I think this should be in the GUI repo.

Moved into https://github.com/bitcoin-core/gui/pull/86.

@MarcoFalke 
> Also, it would be good to add the svg source in case alternative versions need to be created.

Done in https://github.com/bitcoin-core/gui/pull/86.

**Pull request closed without merging**